mesh/mesh.cpp

```diff
@@ -10545,17 +10545,17 @@ GeometricFactors::GeometricFactors(const Mesh *mesh, const IntegrationRule &ir,
    unsigned eval_flags = 0;
    if (flags & GeometricFactors::COORDINATES)
    {
-      X.SetSize(vdim*NQ*NE);
+      X.SetSize(vdim*NQ*NE, Device::GetDeviceTempMemoryType());
```

Comment:
The type of memory use here (regular or temp) should probably be an option in the class.

Reply:
Done!